src/Deck.java

```diff
@@ -0,0 +1,69 @@
+import java.util.Random;
+
+
+public class Deck {
+
+    private Card[] deck;
+    private int currentCard; //index of next card to be dealt
+
+    public Deck() {
+
+        String[] faces = {"2", "3", "4", "5", "6", "7", "8", "9", "10", "Jack", "Queen", "King", "Ace"};
+        String[] suits = {"Spades", "Clubs", "Diamonds", "Hearts"};
+
+        deck = new Card[52];
+        currentCard = 0;
+
+        /**
+         * Building the deck
+         */
+        int suitCounter = 0;
+        while (suitCounter < 4) {
+            int faceNum = 0;
+            while (faceNum < 13) {
+                deck[(faceNum + (suitCounter * 13))] = new Card (
+                        suits[suitCounter],
+                        faces[faceNum],
+                        faceNum+2);
+
+                faceNum++;
+            }
+            suitCounter++;
+        }
+    }
+
+    /**
+     * Printing the deck
+     */
+    public void displayDeck() {
+
+        for (Card card : deck)
+            System.out.println(card);
+
+    }
+
+    public void shuffle() {
+
+        Random mrRandom = new Random();
+        int currentCard = 0;
+        int first = 0;
+        while (currentCard < deck.length) {
```

Comment:
can you explain this shuffle algorithm in plain English? what happen inside this `while loop`?

Reply:
Create random
Current card is used to track how many cards we've been through. Just going to the end of the deck.
"first" is the first card and "second" is the second card. these two will be swapping
while the current card is less than the deck length, then the "first" card is the first of the pair I will use to swap. So I set a temp card equal to the "first" card. Then I used the random to call a "second" card at that random number index. I set the "first" card to equal the "second" and then the "second" equal to the temp (which was the first card).

I'm just swapping cards two at a time with each other. The end result is still a shuffled deck.

Reply:
nice approach,